### PR TITLE
Add RISC-V build to CI

### PR DESCRIPTION
**Pull request by jforissier**
Implementing https://github.com/OP-TEE/optee_os/issues/6704.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -263,6 +263,11 @@ jobs:
           _make PLATFORM=nuvoton
           _make PLATFORM=d06
 
+          export ARCH=riscv
+          export CROSS_COMPILE64="ccache riscv64-linux-gnu-"
```
Comment (etienne-lms) [inline .github/workflows/ci.yml:267]: I suggest you unset `CROSS_COMPILE32` here and change the prelude to arm targets build at line 90, for consistency:
```patch
           # build task
           set -e -v	          set -e -v
           export LC_ALL=C	          export LC_ALL=C
           export PATH=/usr/local/bin:$PATH  # clang
-          export CROSS_COMPILE32="ccache arm-linux-gnueabihf-"arm-linux-gnueabihf-"
-          export CROSS_COMPILE64="ccache aarch64-linux-gnu-"
           export CFG_DEBUG_INFO=n
           (...)
 
           download_plug_and_trust
           download_scp_firmware
 
+          export CROSS_COMPILE32="ccache arm-linux-gnueabihf-"arm-linux-gnueabihf-"
+          export CROSS_COMPILE64="ccache aarch64-linux-gnu-"
+
          _make	          _make
          _make COMPILER=clang

**Review by etienne-lms — COMMENTED** [review]
`Reviewed-by: Etienne Carriere <etienne.carriere@foss.st.com>` for commits:
"riscv: set default cross-compilers (riscv-linux-gnu-gcc and riscv64-linux-gnu-gcc)"
"ldelf: riscv: e64_relocate(): tag sym_idx as __maybe_unused",
"libunw: arm: unwind_arm32.c should be compiled only for Arm",
"libunw: riscv: simplify architecture test" and
"mk/lib.mk: add library to link line only when it does contain objects".

For commit "riscv: set default cross-compilers (riscv-linux-gnu-gcc and riscv64-linux-gnu-gcc)": maybe shorten the header line to _"riscv: set default cross-compilers"_.

**Comment by jforissier** [comment]
Comments addressed, force pushed. I have slightly expanded the explanations in "mk/lib.mk: add library to link line only when it does contain objects" as to why we can't simply guard the inclusion of `lib/lib.mk` with `CFG_UNWIND`.

**Review by etienne-lms — COMMENTED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@foss.st.com>` for commit 
"ci: add RISC-V build (rv64, PLATFORM=virt)".

**Comment by gagachang** [comment]
Hi @jforissier 

We are so grateful to you for adding this work into upstream 😄 

For commits:
"riscv: set default cross-compilers"
"ldelf: riscv: e64_relocate(): tag sym_idx as __maybe_unused"
"libunw: riscv: simplify architecture test"
"ci: add RISC-V build (rv64, PLATFORM=virt)"

Please apply my tag
`Reviewed-by: Alvin Chang <alvinga@andestech.com>`

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Pull request merged**